tools/cli/adminFailoverCommands.go

```diff
@@ -273,6 +200,69 @@ func getRunID(c *cli.Context) string {
 	return ""
 }
 
+func failoverStart(c *cli.Context, params *startParams) {
+	validateStartParams(params)
+
+	targetCluster := params.targetCluster
+	sourceCluster := params.sourceCluster
+	batchFailoverSize := params.batchFailoverSize
+	batchFailoverWaitTimeInSeconds := params.batchFailoverWaitTimeInSeconds
+	workflowTimeout := time.Duration(params.failoverTimeout) * time.Second
+	domains := params.domains
+	drillWaitTime := time.Duration(params.drillWaitTime) * time.Second
+
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	options := cclient.StartWorkflowOptions{
+		ID:                           failovermanager.WorkflowID,
+		WorkflowIDReusePolicy:        cclient.WorkflowIDReusePolicyAllowDuplicate,
+		TaskList:                     failovermanager.TaskListName,
+		ExecutionStartToCloseTimeout: workflowTimeout,
+		Memo: map[string]interface{}{
+			common.MemoKeyForOperator: getOperator(),
+		},
+	}
+	if len(params.cron) > 0 {
+		options.ID = failovermanager.DrillWorkflowID
+		options.CronSchedule = params.cron
+	} else {
+		client := getCadenceClient(c)
+		tcCtx, cancel := newContext(c)
+		defer cancel()
+
+		// block managed failover if an on-going failover drill
+		result := query(tcCtx, client, failovermanager.DrillWorkflowID, "")
+		if result.State == failovermanager.WorkflowRunning {
+			ErrorAndExit("Failed to start failover workflow due to ongoing failover drill", nil)
+		}
+	}
+
+	foParams := failovermanager.FailoverParams{
+		TargetCluster:                  targetCluster,
+		SourceCluster:                  sourceCluster,
+		BatchFailoverSize:              batchFailoverSize,
+		BatchFailoverWaitTimeInSeconds: batchFailoverWaitTimeInSeconds,
+		Domains:                        domains,
+		DrillWaitTime:                  drillWaitTime,
+	}
+	wf, err := client.StartWorkflow(tcCtx, options, failovermanager.WorkflowTypeName, foParams)
+	if err != nil {
+		ErrorAndExit("Failed to start failover workflow", err)
+	}
+	fmt.Println("Failover workflow started")
+	fmt.Println("wid: " + wf.ID)
+	fmt.Println("rid: " + wf.RunID)
+}
+
+func getWorkflowID(c *cli.Context) string {
```

Comment:
nit: shall we make the fn name a bit more specific like `getFailoverWorkflowID`? I think `cli` is a quite large package with many util function, so it may be confusing what this function is doing just by looking at the function name. Same suggestion for the `query` function above.

Reply:
updated